criu/include/cr_options.h

```diff
@@ -75,12 +75,48 @@ enum FILE_VALIDATION_OPTIONS
 	 * This constant indicates that the file validation should be tried with the
 	 * build-ID method by default.
 	 */
-	FILE_VALIDATION_BUILD_ID
+	FILE_VALIDATION_BUILD_ID,
+
+	/*
+	 * This constant indicates that the file validation should be tried with the
+	 * CRC32C checksum method by default.
+	 */
+	FILE_VALIDATION_CHKSM,
+
+	/*
+	 * This constant indicates that the file validation should be done with the
+	 * checksum method using the entire file.
+	 */
+	FILE_VALIDATION_CHKSM_EVERY,
+
+	/*
+	 * This constant indicates that the file validation should be done with the
+	 * checksum method using the first N (The checksum parameter) bytes of the file.
+	 */
+	FILE_VALIDATION_CHKSM_FIRST,
+
+	/*
+	 * This constant indicates that the file validation should be done with the
+	 * checksum method using every Nth (The checksum parameter) bytes of the file.
+	 */
+	FILE_VALIDATION_CHKSM_PERIOD
```

Comment:
Excessive commenting is just force of habit from my course projects. I've changed the comments and changed FILE_VALIDATION_CHKSM_EVERY to FILE_VALIDATION_CHKSM_FULL (Since the argument for it is checksum-full).